Realm/RLMRealm.mm

```diff
@@ -456,52 +448,76 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
     return RLMAutorelease(realm);
 }
 
-+ (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration
-                                 queue:(dispatch_queue_t)queue
-                                 error:(NSError **)error {
+static void *cacheKeyForQueue(dispatch_queue_t queue) {
+    if (queue) {
+        if (queue != dispatch_get_main_queue()) {
+            return (__bridge void *)queue;
+        }
+    }
+    else if (!pthread_main_np()) {
+        return pthread_self();
+    }
+
     // The main thread and main queue share a cache key of `std::numeric_limits<uintptr_t>::max()`
     // so that they give the same instance. Other Realms are keyed on either the thread or the queue.
     // Note that despite being a void* the cache key is not actually a pointer;
     // this is just an artifact of NSMapTable's strange API.
-    void *cacheKey = reinterpret_cast<void *>(std::numeric_limits<uintptr_t>::max());
-    if (queue) {
-        if (queue != dispatch_get_main_queue()) {
-            cacheKey = (__bridge void *)queue;
-        }
+    return reinterpret_cast<void *>(std::numeric_limits<uintptr_t>::max());
+}
+
+static bool copySeedFile(RLMRealmConfiguration *configuration, NSError **error) {
```

Comment:
Is this supposed to be on this PR. It doesn't make sense in the initialSubscriptions` context